src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -67,7 +67,9 @@ public Workspace Workspace
             get
             {
                 Contract.ThrowIfTrue(this.WorkspaceKind == CodeAnalysis.WorkspaceKind.RemoteWorkspace, "Access .Workspace off of a RemoteWorkspace Solution is not supported.");
-                return _state.Workspace;
+#pragma warning disable CS0612 // Type or member is obsolete (TODO: obsolete the property)
+                return _state.Services.WorkspaceServices.Workspace;
```

Comment:
To confirm this still works if we're transplanting a solution object from one Workspace to another with SetCurrentSolution? The behavior we have is "funky" generally here (since we don't refresh everything being held by all the various document/projec states), but we at least still get the core Solution.Workspace property updated, right?

Reply:
Yes, it "works" - we set `SolutionServices`, which point to the workspace.